packages/driver/test/cypress/integration/commands/actions/check_spec.js

```diff
@@ -0,0 +1,1233 @@
+            cy.$$(this).addClass('checked')
+          }
+          , 100)
+        })
+
+        cy.get(':checkbox').invoke('slice', 0, 2).check().should('have.class', 'checked')
+      })
+    })
+
+    describe('events', () => {
+      it('emits click event', (done) => {
+        $('[name=colors][value=blue]').click(() => {
+          done()
+        })
+
+        cy.get('[name=colors]').check('blue')
+      })
+
+      it('emits change event', (done) => {
+        $('[name=colors][value=blue]').change(() => {
+          done()
+        })
+
+        cy.get('[name=colors]').check('blue')
+      })
+
+      it('emits focus event', () => {
+        let focus = false
+
+        $('[name=colors][value=blue]').focus(() => {
+          focus = true
+        })
+
+        cy.get('[name=colors]')
+        .check('blue')
+        .then(() => {
+          expect(focus).to.eq(true)
+        })
+      })
+    })
+
+    describe('errors', () => {
+      beforeEach(() => {
+        Cypress.config('defaultCommandTimeout', 100)
+
+        this.logs = []
+
+        cy.on('log:added', (attrs, log) => {
+          this.lastLog = log
+
+          this.logs.push(log)
+        })
+      })
+
+      it('throws when subject isnt dom', (done) => {
+        cy.on('fail', (err) => {
+          done()
+        })
+
+        cy.noop({}).check()
+      })
+
+      it('throws when subject is not in the document', (done) => {
+        let checked = 0
+
+        const checkbox = $(':checkbox:first').click((e) => {
+          checked += 1
+          checkbox.remove()
+        })
+
+        cy.on('fail', (err) => {
+          expect(checked).to.eq(1)
+          expect(err.message).to.include('cy.check() failed because this element')
+
+          done()
+        })
+
+        cy.get(':checkbox:first').check().check()
+      })
+
+      it('throws when subject isnt a checkbox or radio', (done) => {
+        cy.on('fail', (err) => {
+          expect(err.message).to.include('cy.check() can only be called on :checkbox and :radio. Your subject contains a: <form id="by-id">...</form>')
+
+          done()
+        })
+
+        // this will find multiple forms
+        cy.get('form').check()
+      })
+
+      it('throws when any member of the subject isnt a checkbox or radio', (done) => {
+        cy.on('fail', (err) => {
+          expect(err.message).to.include('cy.check() can only be called on :checkbox and :radio. Your subject contains a: <textarea id="comments"></textarea>')
+
+          done()
+        })
+
+        // find a textare which should blow up
+        // the textarea is the last member of the subject
+        cy.get(':checkbox,:radio,#comments').check()
+      })
+
+      it('throws when any member of the subject isnt visible', (done) => {
+        const chk = cy.$$(':checkbox').first().hide()
+
+        cy.on('fail', (err) => {
+          const {
+            lastLog,
+          } = this
+
+          expect(this.logs.length).to.eq(chk.length + 1)
```

Comment:
`this` with arrow functions, is this maintaining the correct `this`?

Reply:
Yes.